### ISSUE TITLE
Improve feedback from checks for contributors

### ISSUE THREAD

**Issue by ashander** [tskit-dev/msprime#197]
for example in #196 I got feedback from codedcov (which is much appreciated and useful) via a comment which made things seem all good, but didn't get any feedback from the (multiple, failing builds) because they're just checks on github and don't comment. 

I'd suggest a change to avoid this dissonance:

1. disabling codecov comments or
2. adding comments/notifications for failing builds

**Comment by jeromekelleher** [comment]
Following on the discussion at #196, we should change the CodeCov status message to only happen when CI succeeds. This could possibly be done by making the CodeCov upload happen 'on-success'.

**Comment by ashander** [comment]
Yeah that seems the way to go. The bot itself has some configuration possibilities but none to not comment https://docs.codecov.io/docs/pull-request-comments

I'm trying this out [in ftprime (with circle 2.0)](https://github.com/ashander/ftprime/commit/17f2ab57b346825b2d482d1892eaabdd2e587af4)
and will let you know how it goes

**Comment by jeromekelleher** [comment]
> I'm trying this out in ftprime (with circle 2.0) and will let you know how it goes

That would be great, thanks.

**Comment by ashander** [comment]
looks like that works. eg this PR's tests failed and the bot didn't comment https://github.com/ashander/ftprime/pull/26

**Comment by jeromekelleher** [comment]
OK, cool. This will need a bit of a update on the circeci config, and I'm wondering if I should move to 2.0 while I'm at it.

It looks like Circle 2 allows you to break up tests into independent chunks. I would like to run the Python and C tests, and upload their coverage reports only on success. (Ideally only if both succeed I guess, but that might just be over-complicating things.) Is moving to Circle2 worth it?

**Comment by ashander** [comment]
It's working well, but the syntax for the config seems a lot more verbose. The caching system is faster (maybe, or maybe just less used?), but it doesn't seem like your builds are slowed by cache. If things are working smoothly on 1.0, I'd just tweak and keep it there.

I switched because I never had 1.0 dialed and was trying to make builds faster by using conda (instead of a custom docker image) in hopes caching would work better. We needed to use binaries for our dependency on simuPOP. When I decided to make that change, it seemed worth trying out 2.0.
